python/simulation.py

```diff
@@ -98,6 +98,32 @@ def py_v3_to_vec(dims, iterable, is_cylindrical=False):
     else:
         raise ValueError("Invalid dimensions in Volume: {}".format(dims))
 
+class DiffractedPlanewave(object):
+    def __init__(self,
+                 g=None,
+                 axis=None,
```

Comment:
Eventually, we'll have the EigenmodeSource etcetera choose a reasonable default if `axis=None` when the source is added.

(You'll generally want `axis` to lie in the plane of the source, I think.)